### ISSUE TITLE
Warning: Using UNSAFE_componentWillReceiveProps in strict mode is

### ISSUE THREAD

**Issue by carminoplata** [dreyescat/react-rating#146]
Hi everybody,

Does someone deal with this warning?

`Warning: Using UNSAFE_componentWillReceiveProps in strict mode is not recommended and may indicate bugs in your code. See https://reactjs.org/link/unsafe-component-lifecycles for details. 
* Move data fetching code or side effects to componentDidUpdate.
* If you're updating state whenever props change, refactor your code to use memoization techniques or move it to static getDerivedStateFromProps. Learn more at: https://reactjs.org/link/derived-state

Please update the following components: Rating, RatingAPILayer`

**Comment by dreyescat** [comment]
Yep. We need to refactor the component to get rid of this deprecated method.

I'm open to contributions.

The #134 issue was created to track the move towards a controlled component that will let us to get rid of this deprecated lifecycle method.

**Comment by bayramn** [comment]
I still have same issue. How can I resolve it?

**Comment by carlocilento** [comment]
The issue is still present as of January 2022 :/

**Comment by mleister97** [comment]
Issue does still exist

**Comment by BorjaRafolsMartinez** [comment]
Fixed it in here [Selectra-Dev/react-rating#b76396d4c9c1680153817d7869a6d152d1985a90](https://github.com/Selectra-Dev/react-rating/commit/b76396d4c9c1680153817d7869a6d152d1985a90)

**Comment by mariomeraz** [comment]
> Fixed it in here [Selectra-Dev/react-rating#b76396d4c9c1680153817d7869a6d152d1985a90](https://github.com/Selectra-Dev/react-rating/commit/b76396d4c9c1680153817d7869a6d152d1985a90)

amazing , but how do I tell my json to take that fork

**Comment by mariomeraz** [comment]
> Fixed it in here [Selectra-Dev/react-rating#b76396d4c9c1680153817d7869a6d152d1985a90](https://github.com/Selectra-Dev/react-rating/commit/b76396d4c9c1680153817d7869a6d152d1985a90)

yarn add https://github.com/Selectra-Dev/react-rating.git

**Comment by arcasoy** [comment]
Are there any plans to merge those changes^ into this repo?

**Comment by Yvad60** [comment]
The issue still exists July 2022

**Comment by mariomeraz** [comment]
> Are there any plans to merge those changes^ into this repo?

Is Fork, and no have idea...

**Comment by Phasor** [comment]
Any update as of January 2023? Issue is still here.

**Comment by Roger5791** [comment]
Any Update....?
